hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/RatisPipelineUtils.java

```diff
@@ -100,4 +105,45 @@ static void destroyPipeline(DatanodeDetails dn, PipelineID pipelineID,
           true, p.getId());
     }
   }
+
+  static int encodeNodeIdsOfFactorThreePipeline(List<DatanodeDetails> nodes) {
+    if (nodes.size() != HddsProtos.ReplicationFactor.THREE.getNumber()) {
+      return 0;
+    }
+    List<UUID> nodeIds = nodes.stream()
+        .map(DatanodeDetails::getUuid).distinct()
+        .collect(Collectors.toList());
+    nodeIds.sort(new ComparableComparator());
+    // Only for Factor THREE pipeline.
+    return new HashCodeBuilder()
+        .append(nodeIds.get(0).toString())
+        .append(nodeIds.get(1).toString())
+        .append(nodeIds.get(2).toString())
+        .toHashCode();
+  }
+
+  /**
+   * Return first existed pipeline which share the same set of datanodes
+   * with the input pipeline.
+   * @param stateManager PipelineStateManager
+   * @param pipeline input pipeline
+   * @return first matched pipeline
+   */
+  static Pipeline checkPipelineContainSameDatanodes(
+      PipelineStateManager stateManager, Pipeline pipeline) {
+    List<Pipeline> matchedPipelines = stateManager.getPipelines(
+        HddsProtos.ReplicationType.RATIS,
+        HddsProtos.ReplicationFactor.THREE)
+        .stream().filter(p -> !p.equals(pipeline) &&
```

Comment:
Need to check the override of equals() in Pipeline object, the list of nodes may not in the same order. This requires us to use getNodesInOrder instead of getNodes().

Reply:
I will use PipelineID then